### PR TITLE
[Bug](thread-pool) try to fix pv operator of work thread pool

### PR DESCRIPTION
**Pull request by BiteTheDDDDt**
## Proposed changes
I think get_cv is not notified enough

 errCode = 2, detailMessage = (172.21.0.37)[CANCELLED]exchange req success but status isn't ok: [CANCELLED]fail to offer request to the work pool, pool=PriorityThreadPool(name=brpc_heavy, queue_size=10240/10240, active_thread=3/128, total_get_wait_time=132638923328525, total_put_wait_time=350913870127)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by BiteTheDDDDt** [comment]
run buildall

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 48.36 seconds
 stream load tsv:          554 seconds loaded 74807831229 Bytes, about 128 MB/s
 stream load json:         20 seconds loaded 2358488459 Bytes, about 112 MB/s
 stream load orc:          65 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          31 seconds loaded 861443392 Bytes, about 26 MB/s
 insert into select:          29.1 seconds inserted 10000000 Rows, about 343K ops/s
 storage size: 17162484531 Bytes

**Comment by BiteTheDDDDt** [comment]
run buildall

**Comment by BiteTheDDDDt** [comment]
run buildall

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 45.52 seconds
 stream load tsv:          552 seconds loaded 74807831229 Bytes, about 129 MB/s
 stream load json:         21 seconds loaded 2358488459 Bytes, about 107 MB/s
 stream load orc:          65 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          32 seconds loaded 861443392 Bytes, about 25 MB/s
 insert into select:          29.3 seconds inserted 10000000 Rows, about 341K ops/s
 storage size: 17162047625 Bytes

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 47.01 seconds
 stream load tsv:          555 seconds loaded 74807831229 Bytes, about 128 MB/s
 stream load json:         21 seconds loaded 2358488459 Bytes, about 107 MB/s
 stream load orc:          65 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          32 seconds loaded 861443392 Bytes, about 25 MB/s
 insert into select:          29.8 seconds inserted 10000000 Rows, about 335K ops/s
 storage size: 17161979140 Bytes

File: be/src/util/blocking_priority_queue.hpp [hunk 1]
```diff
@@ -154,14 +157,15 @@ class BlockingPriorityQueue {
         }
 
         _queue.push(val);
+        unique_lock.unlock();
```
Comment (yiguolei) [inline be/src/util/blocking_priority_queue.hpp:160]: why add lock for notify?? https://en.cppreference.com/w/cpp/thread/condition_variable/notify_one

Comment (BiteTheDDDDt) [reply be/src/util/blocking_priority_queue.hpp:160]: putting the notify_one before unlock:
notifying thread: notify -> eventually release lock
notified thread: awaken -> attempt to acquire lock and fail -> block until lock available -> acquire lock after 
notifying thread releases it

putting the notify_one after unlock:
notifying thread: notify 
notified thread: awaken -> attempt to acquire lock and succeed

https://stackoverflow.com/questions/55154462/c-condition-variable-notify-one-before-or-after-release-lock

**Comment by BiteTheDDDDt** [comment]
run buildall

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 49.02 seconds
 stream load tsv:          535 seconds loaded 74807831229 Bytes, about 133 MB/s
 stream load json:         20 seconds loaded 2358488459 Bytes, about 112 MB/s
 stream load orc:          64 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          31 seconds loaded 861443392 Bytes, about 26 MB/s
 insert into select:          29.0 seconds inserted 10000000 Rows, about 344K ops/s
 storage size: 17162043359 Bytes

**Pull request closed without merging**